### ISSUE TITLE
sumologicschemaprocessor: allow aggregating attributes with given name patterns 

### ISSUE THREAD

**Issue by aboguszewski-sumo** [SumoLogic/sumologic-otel-collector#866]
## Problem
The processor should have a possibility to aggregate attributes with given name pattern into a single key-value pair. 

This functionality can't be achieved now by the `transform` processor, because some attribute names are created dynamically.

## Example
Given the following list of attributes:
```
[
  "pod_labels_app": "demo",
  "pod_labels_pod_template_hash": "123456#&",
  "pod_labels_foo": "bar"
]
```
let's say that the user wants to map all pods with name `pod_labels_*` to a name `kubernetes.labels`. The result should be:
```
"kubernetes.labels": {
  "app": "demo",
  "pod_template_hash": "123456#&",
  "foo": "bar"
}
```

## Notes

It is assumed that only wildcards (`*`) are supported. If any other kind pattern matching should be supported (eg. regexes), it should be clear what the result names should be.

_Original upstream issue: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/14946_

**Comment by swiatekm-sumo** [comment]
Looking at the implementation questions, I think we should be more specific about what kinds of patterns we support. I see two basic approaches:
1. Only prefixes
    The user specifies a prefix and the suffix is the name of the key in the map. For the example, it'd just be `pod_labels`
2. Full regex matching
    The user specifies a regex with a named capture group, and we use the name of that group.

@rupeshnemade do you need 2? As this is being worked on upstream, I'd like to keep this feature as simple as possible.

**Comment by rupeshnemade** [comment]
@swiatekm-sumo First approaches found to solve the use case as we can map all pod_labels_* values to single key 'labels'.
If 2nd is also doing the same but differently, then may be if you could provide an example then it will be helpful.

**Comment by rupeshnemade** [comment]
@aboguszewski-sumo As of now we need it for labels as thats the only pattern with common prefix. 
Though 2nd approach sounds robust to handle all use cases, I think it needs predefined regex pattern (pod_[abc..]) which we can't predict as labels are very dynamic. But it will be definitely of use in future for complex use case if any besides labels. 
At the moment approach 1 would solve our problem so we can go ahead with it.

**Comment by aboguszewski-sumo** [comment]
@rupeshnemade Thanks, we'll implement the first one then and improve it later if needed.

That example was just made to show how regexes with named groups would work, dynamic names can be easily handled by inserting `.*` anyway, for example `(?P<attr_name>.*)_pod_(.*)` would add also translate the `"schema_pod_h": 3` attribute.

**Comment by aboguszewski-sumo** [comment]
This has been resolved by #871
Hypothetically, this feature could be enabled only for resource attributes - this is a potential enhancement.
I will close the issue in January.
